Azure/azure-cli
# [Storage] `az storage blob upload/upload-batch`: Fix upload batch where it exits on the first upload failure

**Description**

Fix #21593 
Fix upload batch where it exits on the first upload failure, now allows uploading a subset based on precondition

**Testing Guide**


#initial upload
`az storage blob upload -c {container_name} -f "{file_path}" -n {blob_name}`
#this upload should overwrite the first 
`az storage blob upload -c {container_name} -f "{file_path}" -n {blob_name} --if-modified-since {timestamp_before_first_upload}`
#this upload should not work
`az storage blob upload -c {container_name} -f "{file_path}" -n {blob_name} --if-unmodified-since {timestamp_before_first_upload}`

#initial upload
`az storage blob upload-batch -d {container_name} -s "{directory path}"`
#this upload should overwrite all 
`az storage blob upload-batch -d {container_name} -s "{directory path}" --if-modified-since {timestamp_before_first_upload}`
#this upload should not overwrite any
`az storage blob upload-batch -d {container_name} -s "{directory path}" --if-unmodified-since {timestamp_before_first_upload}`

**History Notes**


[Storage] `az storage blob upload/upload-batch`: Make precondition work
[Storage] `az storage blob upload-batch`: No longer exits on the first failure

---

This checklist is used to make sure that common guidelines for a pull request are followed.
